File: 205-isomorphic-strings/205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        table = {}
        for i, char in enumerate(s):
            table[char] = t[i]
        if len(table) != len(set(table.values())):
            return False
        new_list = [table[char] for char in s]
        new_list = ''.join(new_list)
        if new_list == t:
            return True
        return False

File: 205-isomorphic-strings/205-isomorphic-strings.cpp
#include <map>
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char, char> table;
        set<char> values;
        for(int i=0; i< s.size(); ++i){
            table[s[i]] = t[i];
            values.insert(t[i]);
        }
        if (values.size() != table.size()) return false;
        string new_string= "";
        for(int i = 0; i < s.size(); ++i){
            new_string += table[s[i]];
        }
        if(new_string == t) return true;
        return false;
    }
};